app/models/user.rb

```diff
@@ -687,24 +687,27 @@ def badge_count
   end
 
   def featured_user_badges(limit=3)
-    user_highest_tl_badge_id = user_badges
-                                   .where("user_badges.badge_id IN (:tl_badge_ids)",
-                                           tl_badge_ids: Badge.trust_level_badge_ids)
-                                   .maximum(:badge_id)
-                                   .to_i
-    user_badges
-        .group(:badge_id)
-        .where("badge_id >= :highest_tl_badge_id",
-                highest_tl_badge_id: user_highest_tl_badge_id)
-        .select(UserBadge.attribute_names.map { |x| "MAX(user_badges.#{x}) AS #{x}" },
-                'COUNT(*) AS "count"',
-                'MAX(badges.badge_type_id) AS badges_badge_type_id',
-                'MAX(badges.grant_count) AS badges_grant_count')
-        .joins(:badge)
-        .order("CASE WHEN user_badges.badge_id = #{user_highest_tl_badge_id} THEN 1 ELSE 0 END DESC")
-        .order('badges_badge_type_id ASC, badges_grant_count ASC')
-        .includes(:user, :granted_by, { badge: :badge_type }, { post: :topic })
-        .limit(limit)
+    tl_badge_ids = Badge.trust_level_badge_ids
+
+    query = user_badges
+              .group(:badge_id)
+              .select(UserBadge.attribute_names.map { |x| "MAX(user_badges.#{x}) AS #{x}" },
+                      'COUNT(*) AS "count"',
+                      'MAX(badges.badge_type_id) AS badges_badge_type_id',
+                      'MAX(badges.grant_count) AS badges_grant_count')
+              .joins(:badge)
+              .order('badges_badge_type_id ASC, badges_grant_count ASC, badge_id DESC')
+              .includes(:user, :granted_by, { badge: :badge_type }, { post: :topic })
+
+    tl_badge = query.where("user_badges.badge_id IN (:tl_badge_ids)",
+                           tl_badge_ids: tl_badge_ids)
+                    .limit(1)
+
+    other_badges = query.where("user_badges.badge_id NOT IN (:tl_badge_ids)",
+                               tl_badge_ids: tl_badge_ids)
+                        .limit(limit)
```

Comment:
If the user is still on tl0 he will need 3 other badges. If the is tl>0 he need 2.

We could count the length of tl_badge (1 or 0)  and make the appropriate changes but I think it would be less readable.